### PR TITLE
refactor: extract BaseRepository[T] from repository classes

### PR DESCRIPTION
**Pull request by jamescrowley321**
## Summary

- Extracts generic `BaseRepository[T]` into `backend/app/repositories/base.py` with shared `create`, `get`, `update`, `delete`, `commit`, `rollback` methods (PEP 695 type parameter syntax)
- Moves `RepositoryConflictError` from `user.py` to `base.py` — all 7 concrete repos and 22 consuming files updated
- Fixes rollback-on-conflict inconsistency: `user.py` and `provider.py` previously rolled back before raising `RepositoryConflictError`, while other repos did not. All repos now consistently do NOT rollback (service layer owns the transaction boundary)
- All 7 repositories inherit `BaseRepository` and only contain domain-specific methods
- Re-exports all concrete repos from `app.repositories.__init__`
- Extracts shared E2E helpers (`unique_name`, formerly `_unique_name`/`_unique_id`) into `tests/e2e/helpers/api.py`, removing duplication across 7 test files
- Removes `timed_request`/`MAX_API_RESPONSE_MS` from all E2E tests — latency monitoring belongs in OTel, not test assertions
- No service or router layer changes needed — public interfaces unchanged

## Test plan

- [x] `make lint` passes (ruff check + format)
- [x] 1676 unit tests passing (1667 existing + 9 new `test_base_repository.py`)
- [x] New `test_base_repository.py`: generic CRUD, conflict handling, inheritance verification
- [x] Adversarial code review passed (Blind Hunter, Edge Case Hunter, Acceptance Auditor — 0 patch findings)
- [ ] Existing E2E tests pass (`make test-e2e`)
- [ ] New `test_identity_repository_api.py`: full lifecycle chain, conflict handling, concurrent entity operations
- [ ] New `test_identity_repository_ui.py`: authenticated navigation, data display via API+browser, error states

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: backend/tests/e2e/test_identity_repository_api.py [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+
+pytestmark = pytest.mark.skipif(
+    not os.environ.get("DESCOPE_MANAGEMENT_KEY"),
+    reason="DESCOPE_MANAGEMENT_KEY not set",
+)
+
+MAX_API_RESPONSE_MS = 2000
+
+
+def _unique_name(prefix: str) -> str:
```
Comment (jamescrowley321) [inline backend/tests/e2e/test_identity_repository_api.py]: Repeated code across tests needs to be refactored
